### ISSUE TITLE
Provide a way to remove specific parameter from query string

### ISSUE THREAD

**Issue by WiseBird** [angular/angular#17202]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

There is no way to remove some parameters from query string when navigating

**What is the motivation / use case for changing the behavior?**

Imagine you have a global filter that affects content on all pages. When you set e.g. start date on the filter the query string is changed to reflect the change. When you remove it then the corresponding parameter should be removed from the query string.

Currently you can add new query parameter by:
```
router.navigate([], {
    queryParams: {
        startDate: ...,
    },
    queryParamsHandling: 'merge', // because you do not want to erase parameters set by other app's components
}));
```

It would be helpful if you also could remove some of them by something like:
```
router.navigate([], {
    queryParams: {
        startDate: undefined,
    },
    queryParamsHandling: 'merge',
}));
```

* **Angular version:** 4.1.0

**Comment by bossqone** [comment]
@jasonaden Hi, any updates? I've posted possible solution at https://github.com/angular/angular/issues/18463#issuecomment-328993740

**Comment by trotyl** [comment]
Closed by #19733 already.
